appender.go

```diff
@@ -316,6 +318,19 @@ func (a *Appender) flush(ctx context.Context, bulkIndexer *BulkIndexer) error {
 		// Add trace IDs to logger, to associate any per-item errors
 		// below with the trace.
 		logger = logger.With(apmzap.TraceContext(ctx)...)
+	} else if a.otelTracingEnabled() {
+		// NOTE: this is missing transaction type information. How is this conveyed in otel?
+		ctx, span = a.config.OtelTracer.Start(ctx, "docappender.flush", trace.WithAttributes(
+			attribute.Int("documents", n),
+		))
+		defer span.End()
+
+		// Add trace IDs to logger, to associate any per-item errors
+		// below with the trace.
+		logger = logger.With(
+			zap.String("traceId", span.SpanContext().TraceID().String()),
+			zap.String("spanId", span.SpanContext().SpanID().String()),
+		)
```

Comment:
For reviewers: A otel zap bridge exists at https://github.com/uptrace/opentelemetry-go-extra/tree/main/otelzap
This implementation is not compatible with a standard zap logger though, as it wraps the logger in a custom `otelzap.Logger` type. Passing the context also requires to use an additional `.Ctx(ctx)` call that is not supported by `zap.Logger`.
Due to this I chose not to use it an instead provide correlation information as standard otel attributes.